panimg/contrib/oct_converter/readers/e2e.py

```diff
@@ -92,114 +67,89 @@ def __init__(self, filepath):
                 chunk = self.chunk_structure.parse(raw)
 
                 if chunk.type == 11:  # laterality data
-                    raw = f.read(20)
-                    try:
-                        laterality_data = self.lat_structure.parse(raw)
-                        if laterality_data.laterality == 82:
-                            self.laterality = "R"
-                        elif laterality_data.laterality == 76:
-                            self.laterality = "L"
-                    except:
-                        self.laterality = None
+                    self.laterality = self.extract_laterality_data(f)
 
                 if chunk.type == 1073741824:  # image data
-                    raw = f.read(20)
-                    image_data = self.image_structure.parse(raw)
-
                     if chunk.ind == 1:  # oct data
-                        all_bits = [
-                            f.read(2)
-                            for i in range(
-                                image_data.height * image_data.width
-                            )
-                        ]
-                        raw_volume = list(
-                            map(self.read_custom_float, all_bits)
-                        )
-                        image = np.array(raw_volume).reshape(
-                            image_data.width, image_data.height
-                        )
-                        image = 256 * pow(image, 1.0 / 2.4)
                         volume_string = f"{chunk.patient_id}_{chunk.study_id}_{chunk.series_id}"
-                        if volume_string in volume_array_dict.keys():
-                            volume_array_dict[volume_string][
-                                int(chunk.slice_id / 2) - 1
-                            ] = image
-                        else:
-                            # try to capture these additional images
-                            if (
-                                volume_string
-                                in volume_array_dict_additional.keys()
-                            ):
-                                volume_array_dict_additional[
-                                    volume_string
-                                ].append(image)
-                            else:
-                                volume_array_dict_additional[volume_string] = [
-                                    image
-                                ]
-                            # print('Failed to save image data for volume {}'.format(volume_string))
+                        volume_dict.append(volume_string)
+                        chunk_stack.append(start)
+
+            # second pass through MDbData chunks:
+            # - extract OCT image data
+            volume_array_dict = {}
+            for volume in set(volume_dict):
+                volume_array_dict[volume] = [0] * len(
+                    [slice for slice in volume_dict if slice == volume]
+                )
+
+            for pos in chunk_stack:
```

Comment:
I think this second passthrough is not really necessary, right? The decoding done below could also be done in the first loop. Or am I missing something?